Sample/EventPipelines/EventPipelines/EventBus.cs

```diff
@@ -0,0 +1,181 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace EventPipelines
+{
+    public interface IEventBus
+    {
+        Task Publish(object @event, CancellationToken ct);
+    }
+
+    public interface IEventHandler
+    {
+        Type CanHandle { get; }
+
+        ValueTask<object?> Handle(object @event, CancellationToken ct);
+    }
+
+    public interface IEventHandler<in TEvent>: IEventHandler
+    {
+        Type IEventHandler.CanHandle => typeof(TEvent);
+
+        async ValueTask<object?> IEventHandler.Handle(object @event, CancellationToken ct)
+        {
+            await Handle((TEvent)@event, ct);
+            return @event;
+        }
+
+        ValueTask Handle(TEvent @event, CancellationToken ct);
+    }
+
+    public class EventHandlerWrapper<TEvent>: IEventHandler<TEvent>
+    {
+        private readonly Func<TEvent, CancellationToken, ValueTask> handler;
+
+        public EventHandlerWrapper(Func<TEvent, CancellationToken, ValueTask> handler)
+        {
+            this.handler = handler;
+        }
+
+        public EventHandlerWrapper(Action<TEvent> handler)
+        {
+            this.handler = (@event, _) =>
+            {
+                handler(@event);
+                return ValueTask.CompletedTask;
+            };
+        }
+
+        public ValueTask Handle(TEvent @event, CancellationToken ct) =>
+            handler(@event, ct);
+    }
+
+    public interface IEventTransformation<in TEvent, TTransformedEvent> : IEventHandler
+    {
+        Type IEventHandler.CanHandle => typeof(TEvent);
+
+        async ValueTask<object?> IEventHandler.Handle(object @event, CancellationToken ct)
+        {
+            return await Handle((TEvent)@event, ct);
```

Comment:
@kbroniek, thank you for the suggestion. This sample is more focused on the Events Broadcast, so publishing events to subscribers. Do you see potential in that? My intent wasn't to replace MediatR, but indeed that approach could also work for any processing 🤔 Thoughts?

Reply:
OK. It will be nice to add a benchmark, but I know it will consume your time that we don't have these days.
Thought you could implement this as .net events or take advantage of source code generation. It could be fast then.
What if we add multiple handlers in the multi-threading app. Consider using ConcurrentDictionary.

Reply:
Fair points. I added an issue for that: https://github.com/oskardudycz/EventSourcing.NetCore/issues/85. I'll tackle that in a dedicated PR, as this branch is living too long. It's about time to kill it 😉